addons/survey/static/src/js/survey_form.js

```diff
@@ -417,38 +425,37 @@ publicWidget.registry.SurveyFormWidget = publicWidget.Widget.extend(SurveyPreloa
      * @param {Promise} nextScreenPromise
      * @param {Object} options see '_submitForm' for details
      */
-    _nextScreen: function (nextScreenPromise, options) {
-        var self = this;
-
+    _nextScreen: async function (nextScreenPromise, options) {
         var resolveFadeOut;
         var fadeOutPromise = new Promise(function (resolve, reject) {resolveFadeOut = resolve;});
 
         var selectorsToFadeout = ['.o_survey_form_content'];
-        if (options.isFinish && !this.nextScreenResult.has_skipped_questions) {
+        if (options.isFinish && !this.nextScreenResult?.has_skipped_questions) {
             selectorsToFadeout.push('.breadcrumb', '.o_survey_timer');
-            cookie.delete('survey_' + self.options.surveyToken);
+            cookie.delete('survey_' + this.options.surveyToken);
         }
-        self.$(selectorsToFadeout.join(',')).fadeOut(this.fadeInOutDelay, function () {
+        this.$(selectorsToFadeout.join(',')).fadeOut(this.fadeInOutDelay, function () {
             resolveFadeOut();
         });
         // Background management - Fade in / out on each transition
         if (this.options.refreshBackground) {
             $('div.o_survey_background').addClass('o_survey_background_transition');
         }
 
-        var nextScreenWithBackgroundPromise = nextScreenPromise.then(function (result) {
-            self.nextScreenResult = result;
+        const [,result] = await nextScreenPromise;
```

Comment:
Hey, linter seem happy ^^

Sorry thought, this line await should be in the async IIFE so that it can run in parallel with the rest and be awaited in the Promise.all below.
(which by the way could also be rewritten with `await` instead of then)